subway_structure.py

```diff
@@ -1588,15 +1602,23 @@ def add(self, el):
                             self.error('Route in two route_masters', m)
                         self.masters[el_id(m)] = el
             elif el['tags'].get('public_transport') == 'stop_area':
-                warned_about_duplicates = False
-                for m in el['members']:
-                    stop_areas = self.stop_areas[el_id(m)]
-                    if el in stop_areas:
-                        if not warned_about_duplicates:
-                            self.warn('Duplicate element in a stop area', el)
-                            warned_about_duplicates = True
-                    else:
-                        stop_areas.append(el)
+                if el['tags'].get('type') != 'public_transport':
```

Comment:
We can get rid of the excessive indentations and store `el['tags'].get('type')` in a variable:
```python
    def add(self, el):
        if el['type'] == 'relation' and 'members' not in el:
            return

        self.elements[el_id(el)] = el
        if not (el['type'] == 'relation' and 'tags' in el):
            return

        tags_type = el['tags'].get('type')
        if tags_type == 'route_master':
            for m in el['members']:
                if m['type'] != 'relation':
                    continue

                if el_id(m) in self.masters:
                    self.error('Route in two route_masters', m)
                self.masters[el_id(m)] = el

        elif el['tags'].get('public_transport') == 'stop_area':
            if tags_type != 'public_transport':
                self.warn(
                    f"stop_area relation with "
                    f"type={tags_type}, needed type=public_transport",
                    el
                )
                return

            warned_about_duplicates = False
            for m in el['members']:
                stop_areas = self.stop_areas[el_id(m)]
                if el in stop_areas and not warned_about_duplicates:
                    self.warn('Duplicate element in a stop area', el)
                    warned_about_duplicates = True
                else:
                    stop_areas.append(el)
```

Reply:
\+